modules/claims_api/spec/sidekiq/v2/poa_form_builder_job_spec.rb

```diff
@@ -0,0 +1,193 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'pdf_fill/filler'
+
+RSpec.describe ClaimsApi::V2::PoaFormBuilderJob, type: :job do
+  subject { described_class }
+
+  let(:power_of_attorney) { create(:power_of_attorney, :with_full_headers) }
+  let(:poa_code) { 'ABC' }
+
+  before do
+    Sidekiq::Job.clear_all
+  end
+
+  describe 'generating the filled and signed pdf' do
+    context '2122a' do
+      before do
+        power_of_attorney.form_data = {
+          recordConsent: true,
+          consentAddressChange: true,
+          consentLimits: ['DRUG ABUSE', 'SICKLE CELL'],
+          veteran: {
+            serviceBranch: 'ARMY',
+            address: {
+              numberAndStreet: '2719 Hyperion Ave',
+              city: 'Los Angeles',
+              state: 'CA',
+              country: 'US',
+              zipFirstFive: '92264'
+            },
+            phone: {
+              areaCode: '555',
+              phoneNumber: '5551337'
+            }
+          },
+          claimant: {
+            firstName: 'Lillian',
+            middleInitial: 'A',
+            lastName: 'Disney',
+            email: 'lillian@disney.com',
+            relationship: 'Spouse',
+            address: {
+              numberAndStreet: '2688 S Camino Real',
+              city: 'Palm Springs',
+              state: 'CA',
+              country: 'US',
+              zipFirstFive: '92264'
+            },
+            phone: {
+              areaCode: '555',
+              phoneNumber: '5551337'
+            }
+          },
+          representative: {
+            poaCode: poa_code.to_s,
+            type: 'SERVICE ORGANIZATION REPRESENTATIVE',
+            firstName: 'Bob',
+            lastName: 'Representative',
+            organizationName: 'I Help Vets LLC',
+            address: {
+              numberAndStreet: '2719 Hyperion Ave',
+              city: 'Los Angeles',
+              state: 'CA',
+              country: 'US',
+              zipFirstFive: '92264'
+            }
+          }
+        }
+        power_of_attorney.save
+      end
+
+      it 'generates the pdf to match example' do
```

Comment:
I copied this spec and the one on line 151 from V1. I don't think it's that useful of a test. It doesn't actually test the contents of the pdf (which the constructor specs do), it just asserts that classes receive the right method calls and that the original code can be called.

Reply:
yeah, not really following the tests here so I'm glad to trim the fat if we can find ways to do so.

Reply:
Given that the constructor tests the PDFs generated do actually match, testing that same thing here likely is just not required.  I think you are correct this and the one on line 151 probably are not super useful to keep since this would be verified by the constructor tests.

Reply:
I reworked the generate pdf tests to ensure that the form builder classes are getting passed the correct signatures